Add test for `MethodNode::isDeclaration()`

phansys opened the pull request:
Type: bug report
Issue: n/a
Breaking change: no (at least, not yet)

IIUC,  `MethodNode::isDeclaration()` is not working as expected, because methods overriding or implementing parent classes or interfaces are being interpreted as declarations.

The added test tries to expose the described issue.

phansys committed:
Add test for `MethodNode::isDeclaration()`

AJenbo commented:
Are you running the checks locally? Some will only run when others pass but can provide valuable debug info

phansys commented:
> Are you running the checks locally? Some will only run when others pass but can provide valuable debug info

No, I wasn't checking locally based on the pain generated by the old PHP versions required in the `master` branch.

But now, with `3.x` I guess I have no excuse :pray: 
I'll be pushing again after running the checks locally.

Thank you so much!

AJenbo committed:
Merge remote-tracking branch 'upstream/3.x' into method_declaration_3

# Conflicts:
#	tests/php/PHPMD/Node/MethodNodeTest.php

AJenbo committed:
Fix test and fixtures after merge with 3.x

- Copy inheritance fixtures to tests/resources/ (new location)
- Rewrite test to parse all inheritance files together via PDepend
- Add phpstan-ignore comments for magic method calls
- Add phpcs-ignore for intentional non-camel-caps method names
- Apply cs-fixer formatting to fixture files

AJenbo committed:
Merge remote-tracking branch 'upstream/3.x' into method_declaration_3

AJenbo committed:
Clean up MethodNodeTest: remove name-only docblocks, simplify inheritance test
